src/main/scala/rocketchip/Utils.scala

```diff
@@ -15,19 +15,19 @@ import java.nio.{ByteBuffer, ByteOrder}
 
 class RangeManager {
   private var finalized = false
-  private val l = collection.mutable.HashMap[String, Int]()
+  private val l = collection.mutable.ListBuffer[(String, Int)]()
   def add(name: String, element: Int) = { require(!finalized); l += (name -> element) }
   def rangeMap = {
     finalized = true
-    l map {
+    (l map {
       var sum = 0
       x => { sum += x._2; (x._1 -> (sum-x._2, sum)) }
-    }
+    }).toMap
   }
   def range(name: String) = rangeMap(name)
   def print = {
-    rangeMap map { case (name, (start, end)) =>
-      println(s"${name} on port ${start}-${end-1}")
+    rangeMap.toSeq.sortBy(_._2).map { case (name, (start, end)) =>
```

Comment:
Couldn't you just iterate through the ListBuffer again here? It will save you from having to call sortBy. Also, you should definitely be using foreach and not map. Something like

    var start = 0
    l.foreach { case (name, size) =>
        val end = start + size
        println(...)
        start = end
    }